src/Components/Server/src/Circuits/CircuitRegistry.cs

```diff
@@ -297,6 +296,31 @@ private void DisposeTokenSource(DisconnectedCircuitEntry entry)
             }
         }
 
+        internal ValueTask Terminate(string circuitId)
+        {
+            CircuitHost circuitHost;
+            DisconnectedCircuitEntry entry = default;
+            lock (CircuitRegistryLock)
+            {
+                if (ConnectedCircuits.TryGetValue(circuitId, out circuitHost) || DisconnectedCircuits.TryGetValue(circuitId, out entry))
+                {
+                    PermanentDisconnect(circuitHost ?? entry.CircuitHost);
```

Comment:
Is this the only caller of `PermanentDisconnect`? consider just inlining that code here. I'm not sure we want other callers to separate the logic of `PermanentDisconnect` from the dispose logic. 

I'll end up calling this code later in my next work item, and I'd prefer to call `Terminate`.